apps/shell/utils/helpers.js

```diff
@@ -0,0 +1,38 @@
+const path         = require('path');
+const fs           = require('fs');
+const glob         = require('glob');
+const yaml         = require('js-yaml');
+const host_path_rx = '/ssh/([^\\/\\?]+)([^\\?]+)?(\\?.*)?$';
+
+let host_whitelist = new Set;
+
+exports.host_and_dir_from_url = function (url, default_sshhost) {
```

Comment:
We move to much here. The original function was

```
function host_and_dir_from_url(url){
  let match = url.match(host_path_rx),
  hostname = match[1] === "default" ? default_sshhost : match[1],
  directory = match[2] ? decodeURIComponent(match[2]) : null;

  return [hostname, directory];
}
```

With `default_sshhost` as an argument the rest of the whitelist stuff below can remain in app.js. Moving that block of code here properly will look different that what it is here.



It should look pretty close to that.

Reply:
It has to be moved because `host_whitelist` is used in `app.js` during the websocket upgrade process. `host_whitelist` has to be exported by the module.

Line 168 from `app.js`
```js
} else if (!helpers.host_whitelist.has(host)) {
```